### PR TITLE
[Test Only] Add test for Contact Autocomplete widget

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Add test for Contact Autocomplete widget

Before
----------------------------------------
No test for autocomplete widget 

After
----------------------------------------
Test added. Screenshot from test -

image

Comments
----------------------------------------
@KarinG

### CODE REVIEW

File: tests/src/FunctionalJavascript/WebformCivicrmTestBase.php [hunk 1]
```diff
@@ -201,4 +201,38 @@ public function saveCiviCRMSettings() {
...
+   *
+   * @param int $cid
+   */
+  protected function getUFMatchRecord($cid = NULL) {
+    $utils = \Drupal::service('webform_civicrm.utils');
+    return $utils->wf_civicrm_api('UFMatch', 'getsingle', [
+      'sequential' => 1,
+      'contact_id' => $cid ?? "user_contact_id",
+    ]);
+  }
```
Comment (KarinG) [inline tests/src/FunctionalJavascript/WebformCivicrmTestBase.php:220]: Awesome 👍

File: tests/src/FunctionalJavascript/WebformCivicrmTestBase.php [hunk 2]
```diff
@@ -201,4 +201,38 @@ public function saveCiviCRMSettings() {
...
+
+    $element = $this->assertSession()->elementExists('css', "#" . $id);
+    $element->click();
+    $driver->element('xpath', $elementXpath)->postValue(['value' => [$value]]);
+
+    $this->assertSession()->waitForElementVisible('xpath', '//li[contains(@class, "token-input-dropdown")][1]');
+    $this->createScreenshot($this->htmlOutputDirectory . '/autocomplete.png');
+
+    $page->find('xpath', '//li[contains(@class, "token-input-dropdown")][1]')->click();
+  }
```
Comment (KarinG) [inline tests/src/FunctionalJavascript/WebformCivicrmTestBase.php:241]: Oh and a nice example of how to use `xpath` 🙂

**Comment by KarinG** [comment]
Oh we have a test fail. Looking...

**Comment by KarinG** [comment]
Error is:
`CiviCRM_API3_Exception: Expected one UFMatch but found 2`

I ran into this when reworking the Tags test to rootUser. Perhaps what we could do is login as rootUser for all the Edit Element stuff -> and then logout and test as anonymous or login back in as adminUser?

Either that or we figure out why rootUser in the test behaves different compared to uid=1 in our manual testing.

I tried loading the existing contact (belonging to rootUser) in a test after assigning first_name and last_name and I wasn't seeing that on webform load in existing contact element - WHEN logged in as rootUser. As adminUser -> that works as expected.

**Comment by jitendrapurohit** [comment]
Passing now @KarinG 👍

File: tests/src/FunctionalJavascript/WebformCivicrmTestBase.php [hunk 3]
```diff
@@ -74,6 +75,12 @@ protected function setUp() {
       'id' => 'civicrm_webform_test',
       'title' => 'CiviCRM Webform Test',
     ]);
+    //Create civi contact for rootUser.
+    $this->utils->wf_civicrm_api('UFMatch', 'create', [
+      'uf_id' => $this->rootUser->id(),
+      'uf_name' => $this->rootUser->getAccountName(),
+      'contact_id' => $this->createIndividual()['id'],
+    ]);
   }
```
Comment (KarinG) [inline tests/src/FunctionalJavascript/WebformCivicrmTestBase.php:85]: Awesome - somehow on my local testing and on github actions rootUser had no UF Match - totally ok to ensure this manually 👍

Comment (KarinG) [reply tests/src/FunctionalJavascript/WebformCivicrmTestBase.php:85]: That previous commit solves it - awesome

**Comment by KarinG** [comment]
Awesome work and refactoring @jitendrapurohit  🎉 ✅

**Pull request merged**